### PR TITLE
Smoke tests changes and other improvements

### PR DESCRIPTION
**Pull request by spartanezka**
Rewrite smoke tests to pytest

Latency ov metric changes (tokens)

### CODE REVIEW

File: src/cpp_dl_benchmark/rknn_launcher/src/rknn_launcher.cpp [hunk 1]
```diff
@@ -122,6 +122,10 @@ void RKNNLauncher::read(const std::string& model_file, const std::string& weight
     std::ifstream f(model_file, std::ios::binary | std::ios::ate);
     if (!f.is_open()) {
         std::string err_msg = "Can't open " + model_file;
+        // We need to log errors for Android, because uncaught exceptions cause segfault there.
+        // I don't find a way to turn on exceptions catch feature properly on Android, though
+        // documentation states it's possible.
+        // TODO: @i.vikhrev try with newer ndk.
```
Comment (FenixFly) [inline src/cpp_dl_benchmark/rknn_launcher/src/rknn_launcher.cpp:128]: Думаю стоит убрать строку с TODO, сделать issue

File: src/cpp_dl_benchmark/rknn_launcher/src/rknn_launcher.cpp [hunk 2]
```diff
@@ -155,6 +159,11 @@ void RKNNLauncher::read(const std::string& model_file, const std::string& weight
     auto core_mask = ncores_to_rknn_core_mask_map.at(nthreads);
     ret_code = rknn_set_core_mask(rknnContext, core_mask);
     check_throw_error(ret_code, "rknn_set_core_mask failed");
+
+    // rknn allows to set number of cores for batch execution
+    // TODO: enable if we would measure models with batch
```
Comment (FenixFly) [inline src/cpp_dl_benchmark/rknn_launcher/src/rknn_launcher.cpp:164]: Думаю стоит убрать строку с TODO, сделать issue

File: src/cpp_dl_benchmark/tflite_launcher/README.md [hunk 3]
```diff
@@ -191,22 +195,48 @@ so that cmake can find it during configuration step:
...
+
+    - aarch64 Linux:
+
+        ```bash
+        cmake -DCMAKE_TOOLCHAIN_FILE=<dl-benchmark>/src/cpp_dl_benchmark/cmake/aarch64_toolchain.cmake -DCMAKE_BUILD_TYPE=Release -DBUILD_TFLITE_XNNPACK_LAUNCHER=ON -DBUILD_TFLITE_GPU_LAUNCHER=ON -DTENSORFLOW_SRC_DIR=<tensorflow-src-dir> -DTFLITE_BUILD_DIR=<tensorflow-dir>/bazel-bin/tensorflow/lite <dl-benchmark>/src/cpp_dl_benchmark
+        ```
+
+    - aarch64 Android:
+        ```bash
+        cmake -DDCMAKE_SYSTEM_PROCESSOR=aarch64,
```
Comment (FenixFly) [inline src/cpp_dl_benchmark/tflite_launcher/README.md:225]: в файлах выше запятые заменены на '\', стоит сделать также

File: src/cpp_dl_benchmark/tflite_launcher/README.md [hunk 4]
```diff
@@ -191,22 +195,48 @@ so that cmake can find it during configuration step:
...
+        ```bash
+        cmake -DCMAKE_BUILD_TYPE=Release -DBUILD_TFLITE_XNNPACK_LAUNCHER=ON -DTENSORFLOW_SRC_DIR=<tensorflow-src-dir> -DTFLITE_BUILD_DIR=<tensorflow-dir>/bazel-bin/tensorflow/lite <dl-benchmark>/src/cpp_dl_benchmark
+        ```
+
+    Cross-compile for non-x86 platforms:
+
+    - aarch64 Linux:
+
+        ```bash
+        cmake -DCMAKE_TOOLCHAIN_FILE=<dl-benchmark>/src/cpp_dl_benchmark/cmake/aarch64_toolchain.cmake -DCMAKE_BUILD_TYPE=Release -DBUILD_TFLITE_XNNPACK_LAUNCHER=ON -DBUILD_TFLITE_GPU_LAUNCHER=ON -DTENSORFLOW_SRC_DIR=<tensorflow-src-dir> -DTFLITE_BUILD_DIR=<tensorflow-dir>/bazel-bin/tensorflow/lite <dl-benchmark>/src/cpp_dl_benchmark
```
Comment (FenixFly) [inline src/cpp_dl_benchmark/tflite_launcher/README.md:220]: строка слишком длинная, можно разбить символов \\, как сделано выше

Comment (valentina-kustikova) [inline src/cpp_dl_benchmark/tflite_launcher/README.md:220]: Имеет смысл разбить длинные строки, чтобы можно их было просматривать без прокрутки.

File: src/inference/io_adapter.py [hunk 5]
```diff
@@ -529,12 +529,12 @@ def process_output(self, result, log):
         result = result[result_layer_name]
         shapes = self._original_shapes[next(iter(self._original_shapes))]
         c = 3
-        h, w = result.shape[1:]
+        h, w = result.shape[2:]
```
Comment (valentina-kustikova) [inline src/inference/io_adapter.py:532]: Такое изменение точно не сломало обработку результатов сегментационных моделей OpenVINO? Проводилась валидация инференса OpenVINO для какой-либо сегментационной модели, которая ранее была проверена?

File: tests/smoke_test/configs/dl_models/AlexNet_Pytorch.xml [hunk 6]
```diff
@@ -0,0 +1,36 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (valentina-kustikova) [inline tests/smoke_test/configs/dl_models/AlexNet_Pytorch.xml:1]: В названии файла Py**T**orch

**Review by valentina-kustikova — COMMENTED** [review]
Добавила комментарии. Вопрос касательно картинок: насколько допускает лицензия на  эти картинки их выкладывать к нам в репозиторий?

File: tests/smoke_test/configs/dl_models/AlexNet_Pytorch.xml (hunk 6 above)
Comment (spartanezka) [reply tests/smoke_test/configs/dl_models/AlexNet_Pytorch.xml:1]: имя конфига расшифровывается так: <имя модели>_< inference framework >_<некоторые параметры, важные для данного случая>

File: src/cpp_dl_benchmark/rknn_launcher/src/rknn_launcher.cpp (hunk 1 above)
Comment (spartanezka) [reply src/cpp_dl_benchmark/rknn_launcher/src/rknn_launcher.cpp:128]: https://github.com/itlab-vision/dl-benchmark/issues/485

File: src/cpp_dl_benchmark/rknn_launcher/src/rknn_launcher.cpp (hunk 2 above)
Comment (spartanezka) [reply src/cpp_dl_benchmark/rknn_launcher/src/rknn_launcher.cpp:164]: https://github.com/itlab-vision/dl-benchmark/issues/486

File: src/cpp_dl_benchmark/rknn_launcher/README.md [hunk 7]
```diff
@@ -1,6 +1,6 @@
 # RKNN Benchmark
 
-The tool allows to measure deep learning models inference performance on Rockchip NPU with [RKNN API][rknn]. For measurements you'll need a Rockchip device with NPU on board. [RKNPU2][rknn] supports next platforms:
+The tool allows to measure deep learning models inference performance on Rockchip NPU with [RKNN API][RKNN]. For measurements you'll need a Rockchip device with NPU on board. RKNPU2 supports next platforms:
```
[the remark this answers is not in the document]
Comment (spartanezka) [reply src/cpp_dl_benchmark/rknn_launcher/README.md:3]: https://github.com/itlab-vision/dl-benchmark/issues/470#issuecomment-1891602232

File: src/inference/io_adapter.py (hunk 5 above)
Comment (spartanezka) [reply src/inference/io_adapter.py:532]: @asesorov

**Comment by spartanezka** [comment]
> Добавила комментарии. Вопрос касательно картинок: насколько допускает лицензия на эти картинки их выкладывать к нам в репозиторий?

@asesorov

**Comment by asesorov** [comment]
> Добавила комментарии. Вопрос касательно картинок: насколько допускает лицензия на эти картинки их выкладывать к нам в репозиторий?

Картинки взяты из датасетов VOC2012, MS COCO и lyft-udacity-challenge (лицензия CC0)

File: src/inference/io_adapter.py (hunk 5 above)
Comment (asesorov) [reply src/inference/io_adapter.py:532]: Проверка проводилась на модели pspnet-pytorch из OMZ (взята из чеклиста). Похожая проблема с шейпами была в Instance segmentation, как выяснилось, использовался устаревший формат (OpenVINO 2021.3), в связи с чем для Instance segmentation выкачивается legacy-модель из OpenVINO storage. Возможно, следует обновить сегментационную часть io-adapter'а для последней версии OpenVINO.

File: src/cpp_dl_benchmark/tflite_launcher/README.md [hunk 8]
```diff
@@ -191,38 +211,84 @@ so that cmake can find it during configuration step:
...
+TensorFlow Lite launcher supports models in `tflite` formats.
+Some models that require flex delegate may not work on gpu delegate.
 
 <!-- LINKS -->
 [tflite]: https://www.tensorflow.org/lite
 
-[gpu-drivers]: https://dgpu-docs.intel.com/installation-guides/ubuntu/ubuntu-focal.html
+[gpu-drivers]: 
+https://dgpu-docs.intel.com/installation-guides/ubuntu/ubuntu-focal.html
+
```
Comment (valentina-kustikova) [inline src/cpp_dl_benchmark/tflite_launcher/README.md:293]: Надо удалить пустые строки между ссылками.

File: src/inference/io_adapter.py (hunk 5 above)
Comment (valentina-kustikova) [reply src/inference/io_adapter.py:532]: pspnet-pytorch не валидировалась, у нас в массовых экспериментах фигурирует только deeplabv3. Для нее валидация не сломалась?

File: tests/smoke_test/README.md [hunk 9]
```diff
@@ -0,0 +1,28 @@
+# To run smoke test
+
+1. Perform common installation
+   [steps](../../README.md#software-installation).
+
+1. Run benchmark smoke tests using the following command:
+
+   ```
```
Comment (valentina-kustikova) [inline tests/smoke_test/README.md:8]: bash в начале блока с кодом

File: tests/smoke_test/README.md [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+   For caffe tests the following mark should be used: -m=caffe
+
+   For cpp tests the following arguments could be used: --cpp_benchmarks_dir and --openvino_cpp_benchmark_dir
+
+   Pytest performs download and convert steps for test models,
+   runs `inference_benchmark.py` on simple config.
+
+1. Run accuracy tests using the following command:
+
+   ```
```
Comment (valentina-kustikova) [inline tests/smoke_test/README.md:22]: bash в начале блока с кодом

File: tests/smoke_test/configs/dl_models/AlexNet_Pytorch.xml (hunk 6 above)
Comment (valentina-kustikova) [reply tests/smoke_test/configs/dl_models/AlexNet_Pytorch.xml:1]: Я имела ввиду, что буква t маленькая в названии файла, хотя в других названиях сохранен регистр имен фреймворков :)

**Comment by spartanezka** [comment]
Изменения в io adapter были отменены, модель deeplabv3 поставлена на проверку

**Review by valentina-kustikova — DISMISSED** [review]

File: src/cpp_dl_benchmark/tflite_launcher/src/tflite_launcher.cpp [hunk 11]
```diff
@@ -17,15 +17,14 @@
...
-#include <fstream>
-#include <iostream>
-#include <fstream>
 #include <numeric>
 #include <string>
 #include <vector>
 
 #include <nlohmann/json.hpp>
 
+#include <nlohmann/json.hpp>
```
Comment (FenixFly) [inline src/cpp_dl_benchmark/tflite_launcher/src/tflite_launcher.cpp:26]: дублирование строки с include

**Review by FenixFly — CHANGES_REQUESTED** [review]

**Review by asesorov — APPROVED** [review]

**Review by FenixFly — APPROVED** [review]
